backend/tomato/elements/generic.py

```diff
@@ -238,6 +239,46 @@ def after_upload_use(self):
 		self.custom_template = True
 		self.save()
 
+	def checkMigrate(self):
+		#We only migrate if the element is prepared
+		return self.state in [ST_PREPARED]
+	
+	def action_migrate(self,host):
+		
+		from ..host import Host
+		
+		host = Host.objects.filter(name = host)
+		
+		UserError.check(host, code=UserError.NO_RESOURCES, message="Host not found",data={"type": self.TYPE})
+		UserError.check(self.element.host.name != host.name, code=UserError.UNSUPPORTED_ACTION, message="Migrating to the same host not allowed",data={"type": self.TYPE})
+		UserError.check(self.element.checkMigrate(), code=UserError.UNSUPPORTED_ACTION, message="Element can't be migrated",data={"type": self.TYPE})
+		
+	
+		#Download template. Receive download_grant from template and save it to a tempfile?
+		urllib.urlretrieve(self.element.host.grantUrl(self.action("download_grant"), "download"), self.name+"_tmp_image.tar.gz")
+		
+		#Create identical element on new host
+		new_el = host.createElement(self.TYPE, parent=None, attrs=self.attrs, ownerElement=self)
+
+		#Kill old element on old host
+		self.element.action("destroy")
+		
+		#Set new element and save it
+		self.element = new_el
+		self.save()
+		
+		for iface in self.getChildren():
+			iface.action("migrate",host)
```

Comment:
I don't think this can work. How does the interface know the new element that should become its parent?